packages/lb-components/src/components/videoPlayer/TagToolInstanceAdaptor.tsx

```diff
@@ -0,0 +1,308 @@
+/**
+ * @file Implement TagTool interaction through class components.
+ *       Refer to: packages/lb-annotation/src/core/toolOperation/tagOperation.ts
+ * @author Glenfiddish <edwinlee0927@hotmail.com>
+ * @date 2022-05-31
+ */
+
+import React from 'react';
+import { CommonToolUtils, uuid } from '@labelbee/lb-annotation';
+import StepUtils from '@/utils/StepUtils';
+import { jsonParser } from '@/utils';
+import { VideoPlayer } from './index';
+import { VideoTagLayer } from './VideoTagLayer';
+import { IStepInfo } from '@/types/step';
+import _ from 'lodash';
+import type { ObjectString } from './types';
+import { getKeyCodeNumber } from './utils';
+
+export interface IVideoTagInstanceAdaptorProps {
+  imgIndex: number;
+  imgList: any[];
+  pageForward: () => void;
+  pageJump: (page: string) => void;
+  pageBackward: () => void;
+  onMounted: (instance: TagToolInstanceAdaptor) => void;
+  onUnmounted: () => void;
+  step: number;
+  stepList: IStepInfo[];
+}
+
+interface IVideoTagInstanceAdaptorState {
+  tagResult: any[];
+  labelSelectedList: any;
+  valid: boolean;
+}
+
+export class TagToolInstanceAdaptor extends React.Component<
+  IVideoTagInstanceAdaptorProps,
+  IVideoTagInstanceAdaptorState
+> {
+  public fns: { [key: string]: () => void } = {};
+  public videoRef?: HTMLVideoElement;
+  public labelSelectedList: number[] = [];
+
+  constructor(props: IVideoTagInstanceAdaptorProps) {
+    super(props);
+    this.state = {
+      tagResult: [],
+      labelSelectedList: [],
+      valid: true,
+    };
+  }
+
+  public get config() {
+    const stepInfo = StepUtils.getCurrentStepInfo(this.props.step, this.props.stepList);
+    return jsonParser(stepInfo?.config);
+  }
+
+  /** Just implementation, no actual logic */
+  public get history() {
+    return { initRecord: () => {}, pushHistory: () => {} };
+  }
+
+  get currentTagResult() {
+    return this.state.tagResult[0];
```

Comment:
Maybe you need to write more comment like `Since the video tool has no dependency status, the result exists only for the first`? 

And If the `tagResult[0]` is empty, you need to give a default value `{}`